retrieval_service/datastore/providers/alloydb.py

```diff
@@ -53,19 +54,19 @@ def __init__(self, pool: AsyncEngine):
 
     @classmethod
     async def create(cls, config: Config) -> "Client":
+        if cls.__connector == None:
+            cls.__connector = AsyncConnector(refresh_strategy=RefreshStrategy.LAZY)
+
         async def getconn() -> asyncpg.Connection:
-            async with AsyncConnector(
-                refresh_strategy=RefreshStrategy.LAZY
-            ) as connector:
-                conn: asyncpg.Connection = await connector.connect(
-                    # Alloydb instance connection name
-                    f"projects/{config.project}/locations/{config.region}/clusters/{config.cluster}/instances/{config.instance}",
-                    "asyncpg",
-                    user=f"{config.user}",
-                    password=f"{config.password}",
-                    db=f"{config.database}",
-                    ip_type="PUBLIC",
-                )
+            conn: asyncpg.Connection = await cls.__connector.connect(
```

Comment:
Same here, try moving this block of code to inside the `getconn` method.

`datastore/providers/alloydb.py:61: error: Item "None" of "AsyncConnector | None" has no attribute "connect"  [union-attr]`

```suggestion
            if cls.__connector == None:
                cls.__connector = AsyncConnector(refresh_strategy=RefreshStrategy.LAZY)
            conn: asyncpg.Connection = await cls.__connector.connect(
```

Reply:
Awesome. Thanks for the suggestion!